pkg/controller/disruption/disruption.go

```diff
@@ -647,37 +679,57 @@ func (dc *DisruptionController) getExpectedScale(pdb *policy.PodDisruptionBudget
 	for _, pod := range pods {
 		controllerRef := metav1.GetControllerOf(pod)
 		if controllerRef == nil {
-			err = fmt.Errorf("found no controller ref for pod %q", pod.Name)
-			dc.recorder.Event(pdb, v1.EventTypeWarning, "NoControllerRef", err.Error())
-			return
+			dc.recorder.Event(pdb, v1.EventTypeWarning, "NoControllerRef",
+				fmt.Sprintf("found no controller ref for pod %q", pod.Name))
+			continue
 		}
 
 		// If we already know the scale of the controller there is no need to do anything.
 		if _, found := controllerScale[controllerRef.UID]; found {
+			coveredPods = append(coveredPods, pod)
 			continue
 		}
 
 		// Check all the supported controllers to find the desired scale.
-		foundController := false
+		controllerMatch := false
 		for _, finder := range dc.finders() {
-			var controllerNScale *controllerAndScale
-			controllerNScale, err = finder(controllerRef, pod.Namespace)
-			if err != nil {
+			controllerNScale, finderErr := finder(controllerRef, pod.Namespace)
+			// If the gk of the controller is known, but the resource is missing, we are
+			// dealing with an orphaned pod. Create an event, but keep going.
+			if isControllerNotFoundError(finderErr) {
+				dc.recorder.Event(pdb, v1.EventTypeWarning, "ControllerResourceMissing",
+					fmt.Sprintf("controller resource for pod %q missing: %s", pod.Name, finderErr.Error()))
+				controllerMatch = true
```

Comment:
controller is not found - why is the match flag set true ?

Reply:
The idea was to create events that would distinguish between the situation where a pod has a controller of a GK that the controller knows how to handle but the resource no longer exists (i.e. pod is orphaned) and the situation where the GK of the controller is not one the controller knows how to handle. I have realized that the way the way scale subresources are handled means we don't actually know if a resource doesn't exist or it exists but does not implement scale. I have updated the PR so it no longer tries to make this distinction.